### PR TITLE
chore: update release process

### PR DESCRIPTION
**Pull request by DaniPopes**
## Solution

- Automatically generate CHANGELOG.md with [git-cliff](https://github.com/orhun/git-cliff) on a [cargo-release](https://github.com/crate-ci/cargo-release) hook
- Update documentation for the release process and how to contribute

## PR Checklist

-   [ ] Added Tests
-   [ ] Added Documentation
-   [ ] Updated the changelog
-   [ ] Breaking changes

### CODE REVIEW

File: bin/release [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+exec_or_print cargo release commit "${COMMON_FLAGS[@]}" $SIGN_COMMIT
+
+info "publishing crates"
+exec_or_print cargo release publish "${COMMON_FLAGS[@]}" "${WORKSPACE_FLAGS[@]}"
+
+info "tagging commits"
+cargo release tag "${COMMON_FLAGS[@]}" "${WORKSPACE_FLAGS[@]}" $SIGN_TAG
+
+info "pushing commits and tags to remote"
+cargo release push "${COMMON_FLAGS[@]}" "${WORKSPACE_FLAGS[@]}"
```
Comment (gakonst) [inline bin/release:131]: `cargo release public/commit/tag/push` are not commands, is there a mistake here

**Review by gakonst — CHANGES_REQUESTED** [review]
Some changes needed in the release script. I'm still testing this but overall really nice work.

Can you also integrate this https://github.com/cargo-bins/release-pr, so we can start doing releases on cron weekly?

File: bin/release (hunk 1 above)
Comment (prestwich) [reply bin/release:131]: I think it is using this cargo extension? https://github.com/crate-ci/cargo-release

**Comment by gakonst** [comment]
I'm giving https://github.com/MarcoIeni/release-plz a try

File: cliff.toml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+{% else %}\
+    ## [Unreleased]
+{% endif %}\
+{% for group, commits in commits | group_by(attribute="group") %}
+    ### {{ group | title }}
+    {% for commit in commits %}
+        - {% if commit.breaking %}[**breaking**] {% endif %}{{ commit.message | upper_first | split(pat="\\n") | first }}\
+    {% endfor %}
+{% endfor %}\n
+"""
```
Comment (gakonst) [inline cliff.toml:33]: any chance we can also add the PR author in the changelog?

**Review by gakonst — CHANGES_REQUESTED** [review]

File: bin/release (hunk 1 above)
Comment (gakonst) [reply bin/release]: Oh I was on v0.20 looks like latest has more stuff!

**Comment by gakonst** [comment]
OK let's send it, updated my cargo-release command. Will revise with release-plz in the near future.

**Pull request merged**